parlai/tasks/friends/agents.py

```diff
@@ -89,23 +100,80 @@ def setup_data(self, datafile):
                 if (
                     self.character == 'All' and speaker in self.characters
                 ) or speaker == self.character:
+                    text, label, speakers, hasAddedSpeaker = self._get_message_fields(
+                        text, speaker, speakers, prev_context
+                    )
+                    _speakers = speakers[:]
```

Comment:
curious, why do you do this?

Reply:
Pass by reference vs pass by value
If I don't do this then every example becomes the same as the last example, containing all the speakers in the entire episode  (I ran into this bug)